cartographer/sensor/test_helpers.h

```diff
@@ -29,6 +36,47 @@ MATCHER_P(Near, point, std::string(negation ? "Doesn't" : "Does") + " match.") {
   return arg.isApprox(point, 0.001f);
 }
 
+namespace test {
+
+typedef std::tuple<int, std::string, common::Time> CollatorOutput;
+
+struct CollatorInput {
+  static CollatorInput CreateImuData(int trajectory_id, std::string sensor_id,
+                                     int time) {
+    return CollatorInput{
+        trajectory_id,
+        MakeDispatchable(sensor_id, ImuData{common::FromUniversal(time)})};
+  }
+  static CollatorInput CreateTimedPointCloudData(int trajectory_id,
+                                                 std::string sensor_id,
+                                                 int time) {
+    return CollatorInput{
+        trajectory_id,
+        MakeDispatchable(
+            sensor_id,
+            TimedPointCloudData{
+                common::FromUniversal(time), Eigen::Vector3f::Zero(), {}})};
+  }
+  static CollatorInput CreateOdometryData(int trajectory_id,
+                                          std::string sensor_id, int time) {
+    return CollatorInput{
+        trajectory_id,
+        MakeDispatchable(sensor_id,
+                         OdometryData{common::FromUniversal(time),
+                                      transform::Rigid3d::Identity()})};
+  }
+  void MoveToCollator(CollatorInterface* collator) {
+    expected_output =
+        CollatorOutput(trajectory_id, data->GetSensorId(), data->GetTime());
```

Comment:
It's slightly unusual that you set the output here, although you could have set it at construction time. Then you could even declare it const.

Reply:
Done